### PR TITLE
`azurerm_service_plan` - update the limitation of `maximum_elastic_worker_count`

### LINKED ISSUE #18128: azurerm_service_plan with "sku_name = WS1" does not support "maximum_elastic_worker_count"
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.7

### AzureRM Provider Version

3.19.1

### Affected Resource(s)/Data Source(s)

azurerm_service_plan

### Terraform Configuration Files

```hcl
### FAILED - AzureRM Provider 3.19.1 ###

resource "azurerm_service_plan" "svcplan" {
  name                         = "new-std-la-plan"
  location                     = data.azurerm_resource_group.rg.location
  resource_group_name          = data.azurerm_resource_group.rg.name
  os_type                      = "Windows"
  sku_name                     = "WS1"
  maximum_elastic_worker_count = 5

  tags = local.tags
}

### SUCCEEDED - AzureRM version 2.99.0 ###

resource "azurerm_app_service_plan" "svcplan" {
  name                         = "old-std-la-plan"
  location                     = data.azurerm_resource_group.rg.location
  resource_group_name          = data.azurerm_resource_group.rg.name
  kind                         = "elastic"
  maximum_elastic_worker_count = 5

  sku {
    tier = "WorkflowStandard"
    size = "WS1"
  }

  tags = local.tags
}
```

### Debug Output/Panic Output

```shell
NA
```

### Expected Behaviour

The `azurerm_service_plan` resource (azurerm 3.19.1) with `sku_name` = `WS1` should be able to define `maximum_elastic_worker_count`.

We validated that we could manually set the "maximum burst" count in the Azure Portal UI after provisioning `azurerm_service_plan` with sku_name = WS1 **without** `maximum_elastic_worker_count`. This would confirm that the sku WS1 does support "scale out" or "elastic" functionality.

image

######################################

When using the `azurerm_app_service_plan` resource (azurerm 2.99.0) with `sku size` = `WS1` and  `kind` = `elastic` the `maximum_elastic_worker_count` could be defined.

```
resource "azurerm_app_service_plan" "svcplan" {
  name                         = "old-std-la-plan"
  location                     = data.azurerm_resource_group.rg.location
  resource_group_name          = data.azurerm_resource_group.rg.name
  kind                         = "elastic"
  maximum_elastic_worker_count = 5

  sku {
    tier = "WorkflowStandard"
    size = "WS1"
  }
```

### Actual Behaviour
```
Error: `maximum_elastic_worker_count` can only be specified with Elastic Premium Skus

  with module.tf-az-logicapps.azurerm_service_plan.svcplan,
  on .terraform\modules\tf-az-logicapps\main.tf line 70, in resource "azurerm_service_plan" "svcplan":
  70: resource "azurerm_service_plan" "svcplan" {

`maximum_elastic_worker_count` can only be specified with Elastic Premium Skus
```

### Steps to Reproduce

The following will fail during `apply`

```
resource "azurerm_service_plan" "svcplan" {
  name                         = "new-std-la-plan"
  location                     = data.azurerm_resource_group.rg.location
  resource_group_name          = data.azurerm_resource_group.rg.name
  os_type                      = "Windows"
  sku_name                     = "WS1"
  maximum_elastic_worker_count = 5

  tags = local.tags
}
```

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ziyeqf**
Update the limitation of `maximum_elastic_worker_count ` in `azurerm_service_plan` according to the [document ](https://docs.microsoft.com/en-us/azure/app-service/overview-hosting-plans) to fix #18128

Test
---
```
TF_ACC=1 go test -v  ./internal/services/appservice -run=TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku -timeout=60m

=== RUN   TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku
=== RUN   TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku/WS1
=== PAUSE TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku/WS1
=== RUN   TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku/EP1
=== PAUSE TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku/EP1
=== CONT  TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku/WS1
=== CONT  TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku/EP1
--- PASS: TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku (0.00s)
    --- PASS: TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku/WS1 (293.23s)
    --- PASS: TestAccServicePlan_maxElasticWorkerCountForAllSupportedSku/EP1 (301.18s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    301.193s
```

### CODE REVIEW

File: internal/services/appservice/service_plan_resource_test.go [hunk 1]
```diff
@@ -314,7 +314,7 @@ resource "azurerm_service_plan" "test" {
   name                         = "acctest-SP-%[1]d"
   resource_group_name          = azurerm_resource_group.test.name
   location                     = azurerm_resource_group.test.location
-  sku_name                     = "EP1"
+  sku_name                     = "WS1"
```
Comment (magodo) [inline internal/services/appservice/service_plan_resource_test.go:317]: Can we test for both of skus?

File: internal/services/appservice/service_plan_resource.go [hunk 2]
```diff
@@ -185,8 +185,8 @@ func (r ServicePlanResource) Create() sdk.ResourceFunc {
 			}
 
 			if servicePlan.MaximumElasticWorkerCount > 0 {
-				if !strings.HasPrefix(servicePlan.Sku, "EP") && !strings.HasPrefix(servicePlan.Sku, "PC") {
-					return fmt.Errorf("`maximum_elastic_worker_count` can only be specified with Elastic Premium Skus")
+				if strings.HasPrefix(servicePlan.Sku, "B") || strings.HasPrefix(servicePlan.Sku, "F") {
```
Comment (magodo) [inline internal/services/appservice/service_plan_resource.go:188]: Per https://github.com/hashicorp/terraform-provider-azurerm/blob/dce4255bb81d572ce8476d37938b304f2775d128/internal/services/appservice/helpers/service_plan.go#L54-L65, have you tested the other skus and confirmed that they all support `maximum_elastic_worker_count`?

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

File: internal/services/appservice/service_plan_resource.go (hunk 2 above)
Comment (ziyeqf) [reply internal/services/appservice/service_plan_resource.go]: After testing, I think it's better to just add `WS` to allow list to scale out.

**Review by katbyte — APPROVED** [review]
LGTM 🌙  but we have a test failure:
image

**Comment by ziyeqf** [comment]
Hi @katbyte, thanks for the remind, while it seems to be a time-out error, it's not related to this modification while I tried and failed on the main branch too.

**Review by katbyte — APPROVED** [review]
👍  - @ziyeqf can you open a PR to increase the default timeouts then?

otherwise LGTM 🚜

**Comment by ziyeqf** [comment]
@katbyte tried with 10 hours' time-out limitation but it just get stuck in waiting for service. a further investigation needed.

**Pull request merged**